src/playlist-controller.js

```diff
@@ -2211,6 +2220,99 @@ export class PlaylistController extends videojs.EventTarget {
     }
   }
 
+  /**
+   * Add, update, or delete playlists and media groups for
+   * the pathway clones for HLS Content Steering.
+   *
+   * See https://datatracker.ietf.org/doc/draft-pantos-hls-rfc8216bis/
+   *
+   * NOTE: Pathway cloning does not currently support the `PER_VARIANT_URIS` and
+   * `PER_RENDITION_URIS` as we do not handle `STABLE-VARIANT-ID` or
+   * `STABLE-RENDITION-ID` values.
+   */
+  handlePathwayClones_() {
+    const main = this.main();
+    const playlists = main.playlists;
+    const currentPathwayClones = this.contentSteeringController_.currentPathwayClones;
+    const nextPathwayClones = this.contentSteeringController_.nextPathwayClones;
+
+    currentPathwayClones.forEach((clone) => {
+      const newClone = nextPathwayClones.find(c => c.ID === clone.ID);
+
+      // Delete the old pathway clone.
+      if (!newClone) {
+        this.mainPlaylistLoader_.updateOrDeleteClone(clone);
+        this.contentSteeringController_.excludePathway(clone.ID);
+      }
+    });
```

Comment:
I see that `PathwayClone` entity has an ID, I would suggest using Map instead of Array. It will be easier to manipulate

Reply:
Good call, I updated this to use `Map`s in my most recent commit